Security/Core/Authentication/Provider/OAuthProvider.php

```diff
@@ -79,6 +79,13 @@ public function authenticate(TokenInterface $token)
         /* @var OAuthToken $token */
         $resourceOwner = $this->resourceOwnerMap->getResourceOwnerByName($token->getResourceOwnerName());
 
+        if($token->isExpired()){
```

Comment:
You need to fix CS: https://travis-ci.org/hwi/HWIOAuthBundle/jobs/281949145#L604-L605

Reply:
How to inject token storage service in oauth provider without using container...
Do you have any idea ?

Reply:
You need to modify `Resources/config/oauth.xml` file: https://github.com/hwi/HWIOAuthBundle/blob/master/Resources/config/oauth.xml#L98
to:
```xml
<service id="hwi_oauth.authentication.provider.oauth" class="%hwi_oauth.authentication.provider.oauth.class%" public="false">
    <argument type="service" id="security.token_storage" />
</service>
```
Along with this you need to add `__construct()` etc. in that class.

Reply:
You need to modify `Resources/config/oauth.xml` file: https://github.com/hwi/HWIOAuthBundle/blob/master/Resources/config/oauth.xml#L98
to:
```xml
<service id="hwi_oauth.authentication.provider.oauth" class="%hwi_oauth.authentication.provider.oauth.class%" public="false">
    <argument type="service" id="security.token_storage" />
</service>
```
Along with this you need to add `__construct()` etc. in that class.

Reply:
But OAuthProvider class seems some of parameters directly passed by symfony ...
Is it possible to overwrite it...